networkit/cpp/independentset/Luby.cpp

```diff
@@ -5,17 +5,23 @@
  *      Author: Christian Staudt
  */
 
+#include <atomic>
 #include <networkit/auxiliary/Log.hpp>
 #include <networkit/auxiliary/Random.hpp>
 #include <networkit/independentset/Luby.hpp>
 
 namespace NetworKit {
 
 std::vector<bool> Luby::run(const Graph &G) {
-
+    if (G.isDirected()) {
+        throw std::runtime_error("current implementation can only deal with undirected graphs");
+    }
     std::vector<bool> I(G.numberOfNodes(), false); // independent set $I = \emptyset$
     // instead of pruning the graph, store here whether a node in G is still in G'
-    std::vector<bool> V(G.numberOfNodes(), true);
+    std::vector<std::atomic<bool>> V(G.numberOfNodes());
```

Comment:
Since the new version uses `critical` for sections with potential data races, you should revert the changes with `std::atomic` (e.g. use `std::vector<bool>` instead of `std::vector<std::atomic<bool>>`. (Same for all occurrences).